[quant][graphmode][fx] Add test for ResnetBase

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#48939 [quant][graphmode][fx] Add test for ResnetBase**

Summary:
Add numerical test for fx graph mode for resnet base, comparing with eager mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25375342](https://our.internmc.facebook.com/intern/diff/D25375342)

jerryzh168 committed:
[quant][graphmode][fx] Add test for ResnetBase

Summary:
Add numerical test for fx graph mode for resnet base, comparing with eager mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add test for ResnetBase"


Summary:
Add numerical test for fx graph mode for resnet base, comparing with eager mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25375342](https://our.internmc.facebook.com/intern/diff/D25375342)

[ghstack-poisoned]

supriyar commented on test/quantization/test_quantize_fx.py:
```diff
@@ -2189,6 +2197,57 @@ def _test_model_impl(
+        graph_out = graph(data[0][0])
+        self.assertEqual(eager_out, graph_out)
+
+    @skipIfNoFBGEMM
```
Should we test this for both qengine as well?

supriyar commented on test/quantization/test_quantize_fx.py:
```diff
@@ -2189,6 +2197,57 @@ def _test_model_impl(
+            graph.eval()
+            calibrate_or_train = test_only_eval_fn
+            data = self.img_data_2d
+        else:
```
Can we check for `qat` type here and assert if type is set to dynamic?

supriyar reviewed (approved):
lgtm

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add test for ResnetBase"


Summary:
Add numerical test for fx graph mode for resnet base, comparing with eager mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25375342](https://our.internmc.facebook.com/intern/diff/D25375342)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add test for ResnetBase"


Summary:
Add numerical test for fx graph mode for resnet base, comparing with eager mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25375342](https://our.internmc.facebook.com/intern/diff/D25375342)

[ghstack-poisoned]
